### PR TITLE
Implement method to add node into dataflow

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: test/compile/data/dataflow.test.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import {assert} from 'chai';
 import {DataFlowNode} from '../../../src/compile/data/dataflow';
 
 describe('compile/data/dataflow', function() {
-  describe('DataFlowNode', function() {
+  describe.only('DataFlowNode', function() {
```
Comment (kanitw) [inline test/compile/data/dataflow.test.ts]: describe.only **

File: src/compile/data/dataflow.ts [hunk 2]
```diff
@@ -69,6 +69,16 @@ export class DataFlowNode {
     this._parent.removeChild(this);
   }
 
+  /**
+   * Insert another node as a parent of this node.
+   */
+  public insertAsParentOf(other: DataFlowNode) {
```
Comment (kanitw) [inline src/compile/data/dataflow.ts]: other => child?

File: src/compile/data/dataflow.ts [hunk 3]
```diff
@@ -69,6 +69,16 @@ export class DataFlowNode {
...
+    const parent = other.parent;
+    parent.removeChild(this);
```
Comment (kanitw) [inline src/compile/data/dataflow.ts]: removeChild(other) ?

File: test/compile/data/dataflow.test.ts (hunk 1 above)
Comment (domoritz) [inline test/compile/data/dataflow.test.ts]: Fixed

File: src/compile/data/dataflow.ts (hunk 2 above)
Comment (domoritz) [inline src/compile/data/dataflow.ts]: Done

File: src/compile/data/dataflow.ts (hunk 3 above)
Comment (domoritz) [inline src/compile/data/dataflow.ts]: Done and force pushed. Also added test.

**Comment by kanitw** [comment]
Not sure why the other test doesn't pass.

**Comment by domoritz** [comment]
Needs to be restarted. Same issue that I reported earlier.

**Comment by domoritz** [comment]
Feel free to merge. The issues are just in the build system.

**Pull request merged**